src/lib/security/selinux_config.rb

```diff
@@ -0,0 +1,263 @@
+    #
+    # @note it does not write the changes when running in installation mode, where only sets the
+    #   kernel params in memory, since the Yast::Bootloader.Write will be performed at the end of
+    #   installation.
+    #
+    # @see Yast::Bootloader#modify_kernel_params
+    #
+    # @return [Boolean] false if mode is not set or nothing changed; true otherwise
+    def save
+      return false if mode.nil?
+
+      changed = Yast::Bootloader.modify_kernel_params(**mode.options)
+      changed = Yast::Bootloader.Write if changed && !Yast::Mode.installation
+      changed
+    end
+
+    private
+
+    # Path to the SELinux getenforce command
+    GETENFORCE_PATH = "/usr/sbin/getenforce".freeze
+    private_constant :GETENFORCE_PATH
+
+    # Proposes a mode based on `selinux_mode` value set in the control file
+    #
+    # If mode is found, it calls #save for setting it.
+    #
+    # @return [Mode] the configured SELinux mode after trying the proposal
+    def propose!
+      id = Yast::ProductFeatures.GetFeature("globals", "selinux_mode").to_sym
+
+      if Mode.find(id)
+        log.info "Proposing the `#{id}` SELinux mode"
+        self.mode = id
+        save
+      else
+        log.info "Unknown `#{id}` SELinux mode. Skipping the proposal."
+      end
+
+      self.mode
+    end
+
+    # Returns the configured SELinux mode according to params in kernel command line
+    #
+    # @see #mode_from_kernel_params
+    #
+    # @return [Symbol] the mode identifier
+    def configured_mode
+      Mode.match(mode_from_kernel_params)
+    end
+
+    # Returns the SELinux configuration based on options set in the kernel command line
+    #
+    # @return [Symbol] the mode identifier
+    def mode_from_kernel_params
+      params = Mode.keys.flat_map do |key|
+        value = Array(read_param(key)).last
+        next if value == :missing
+
+        [key, value]
+      end
+
+      Hash[*params.compact]
+    end
+
+    # Convenience method to read a value from kernel parameters
+    #
+    # @see Yast::Bootloader#kernel_param
+    def read_param(key)
+      Yast::Bootloader.kernel_param(:common, key.to_s)
+    end
+
+    # Model that represents a SELinux mode
+    class Mode
+      extend Yast::I18n
+      include Yast::I18n
+
+      # @return [Symbol] the id representing the mode
+      attr_reader :id
+      alias_method :to_sym, :id
+
+      # @return [String] the human readable name to represent the mode
+      attr_reader :name
+      alias_method :to_human_string, :name
+
+      # @return [Hash{String=><String, :missing>}] options for setting the mode via kernel params
+      attr_reader :options
+
+      # Returns all known SELinux modes
+      #
+      # @return [Array<Mode>]
+      def self.all
+        ALL.dup
+      end
+
+      # Returns all known keys for setting a SELinux mode via kernel command line
+      #
+      # @return [Array<String>]
+      def self.keys
```

Comment:
I prefer better method name like `kernel_options` or something like it. Otherwise it looks like some mode keys, which is a bit strange.